tests/src/JIT/Regression/JitBlue/GitHub_8170/GitHub_8170.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+// This test checks for proper behavior w.r.t. overflow for expressions of the form `val % constant` where `val` is an
+// unsigned long and `constant` is a 32- or 64-bit integer constant in the range [2,0x3fffffff]. These expressions
+// should never produce an overflow exception.
+
+static class C
+{
+	[MethodImpl(MethodImplOptions.NoInlining)]
+	static uint M(ulong l)
+	{
+		return (uint)(l % 10000000);
+	}
+
+	static int Main()
+	{
+		return (int)M(ulong.MaxValue);
```

Comment:
Yeah, that's what I get from copy/pasting my condensed repro without re-reading it :(